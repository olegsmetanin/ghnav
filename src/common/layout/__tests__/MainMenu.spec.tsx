import * as React from 'react'

/* eslint-env jest */
import { MainMenu } from '../MainMenu'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

describe('MainMenu', () => {
  it('renders menus', () => {
    const tree = mount(<MainMenu>Hello</MainMenu>)
    const menu = tree.find('[data-test-id="mobile-menu-icon"]').first()
    expect(menu.exists()).toBe(true)
    menu.simulate('click')
    const profile = tree.find('[data-test-id="menu-profile-item"]').first()
    expect(profile.exists()).toBe(true)
    profile.simulate('click')
    const subprofile = tree
      .find('[data-test-id="menu-profile-profile-item"]')
      .first()
    expect(subprofile.exists()).toBe(true)
    subprofile.simulate('click')
  })

  it('renders with different width', () => {
    const component = renderer.create(<MainMenu>Hello</MainMenu>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
