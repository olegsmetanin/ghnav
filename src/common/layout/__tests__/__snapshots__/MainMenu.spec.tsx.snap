// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu renders with different width 1`] = `
<div
  className="PrimarySearchAppBar-root-1"
>
  <header
    className="MuiPaper-root-20 MuiPaper-elevation4-26 MuiAppBar-root-11 MuiAppBar-positionFixed-12 MuiAppBar-colorPrimary-18 mui-fixed"
  >
    <div
      className="MuiToolbar-root-47 MuiToolbar-dense-50 MuiToolbar-gutters-48"
    >
      <button
        aria-label="Open drawer"
        className="MuiButtonBase-root-57 MuiIconButton-root-51 MuiIconButton-colorInherit-52 PrimarySearchAppBar-menuButton-3"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-56"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-60"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-78"
        />
      </button>
      Hello
      <div
        className="PrimarySearchAppBar-grow-2"
      />
      <div
        className="PrimarySearchAppBar-sectionDesktop-9"
      >
        <button
          className="MuiButtonBase-root-57 MuiIconButton-root-51 MuiIconButton-colorInherit-52"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-56"
          >
            <span
              className="MuiBadge-root-69"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-60"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
              <span
                className="MuiBadge-badge-70 MuiBadge-colorSecondary-72"
              >
                4
              </span>
            </span>
          </span>
          <span
            className="MuiTouchRipple-root-78"
          />
        </button>
        <button
          className="MuiButtonBase-root-57 MuiIconButton-root-51 MuiIconButton-colorInherit-52"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-56"
          >
            <span
              className="MuiBadge-root-69"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-60"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                />
              </svg>
              <span
                className="MuiBadge-badge-70 MuiBadge-colorSecondary-72"
              >
                17
              </span>
            </span>
          </span>
          <span
            className="MuiTouchRipple-root-78"
          />
        </button>
        <button
          aria-haspopup="true"
          aria-owns={null}
          className="MuiButtonBase-root-57 MuiIconButton-root-51 MuiIconButton-colorInherit-52"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-56"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-60"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-78"
          />
        </button>
      </div>
      <div
        className="PrimarySearchAppBar-sectionMobile-10"
      >
        <button
          aria-haspopup="true"
          className="MuiButtonBase-root-57 MuiIconButton-root-51 MuiIconButton-colorInherit-52"
          data-test-id="mobile-menu-icon"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-56"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-60"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-78"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
